stac_generator/raster_data/schema.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import json
+from datetime import UTC, datetime
+from typing import Any
+
+from pydantic import BaseModel, Field, model_validator
+
+from stac_generator.base.schema import SourceConfig
+
+
+class BandInfo(BaseModel):
+    """Band information for raster data"""
+
+    name: str
+    wavelength: float | None = Field(default=None)  # Can be float or None
+    nodata: float | None = Field(default=0)  # Default nodata value
+    data_type: str | None = Field(default="uint16")  # Default data type for raster band
+
+    @model_validator(mode="before")
+    @classmethod
+    def parse_wavelength(cls, data: Any) -> Any:
+        """Handle 'no band specified' case"""
+        if isinstance(data, dict) and data.get("wavelength") == "no band specified":
+            data["wavelength"] = None
+        return data
+
+
+class RasterSourceConfig(SourceConfig):
+    """Configuration for raster data sources"""
+
+    location: str
+    """Location of the raster file"""
+    epsg: int
+    """EPSG code for the raster's coordinate reference system"""
+    collection_date: str  # Changed to str to handle the input format
```

Comment:
Not really sure if `collection_date` here means the datetime when data was acquired or the datetime field of the collection. Either way, these should not be used and instead the fields `datetime`, `start_datetime`, and `end_datetime` native to STAC metadata should be used instead.